README.md

```diff
@@ -1,6 +1,33 @@
-IPFS Test Swarm Container
-=========================
+IPFS Test Network Container
+===========================
 
+Run an isolated, private IPFS network in a Docker container.
+
+The container runs two IPFS nodes that share a [private IPFS
+network][ipfs-private-network]. This allows for isolation from the main IPFS
+network and fast and reliable responses, especially with IPNS.
+
+[ipfs-private-network]: https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#private-networks
+
+Usage
+-----
+
+```
+docker build . -t ipfs-test-network
+docker run \
+  --volume ipfs-test-data:/data \
+  --publish 4001:4001 \
+  --publish 5001:5001 \
+  ipfs-test-network
+```
+
+You can now talk to a node in the test network through the `ipfs` CLI.
+
+```
+ipfs swarm peers
```

Comment:
This needs to be run _inside_ the container? Or is there some setup which would allow `ipfs` on the host to join the private network (can't seem to be able to figure out how)?

Reply:
Yeah, I also am curious about the reasoning for using containers. Isn't it enough to just set the `IPFS_PATH`?

Reply:
> This needs to be run _inside_ the container? Or is there some setup which would allow `ipfs` on the host to join the private network (can't seem to be able to figure out how)?

This can be run from outside of the container if the API port is published.

> Yeah, I also am curious about the reasoning for using containers. Isn't it enough to just set the `IPFS_PATH`?

Theoretically we could use the two bash scripts to spin up a test network. I just find that a container is a nice way to package that.

Also note that we need at least two IPFS nodes for IPNS to work. Controlling two detached processes from bash is not easy. This is why this also depends on `tini` for PID 1.

Reply:
Docker is fine. But still, no luck:

```shell
$ docker run --rm --volume $(pwd)/ipfs-test-data:/data --publish 4001:4001 --publish 5001:5001 ipfs-test-network
```

```shell
$ IPFS_PATH=ipfs-local ipfs init
$ IPFS_PATH=ipfs-local ipfs swarm peers
Error: This command must be run in online mode. Try running 'ipfs daemon' first.
Use 'ipfs swarm peers --help' for information about this command
```

If I run `ipfs daemon` with the local `IPFS_PATH`, the output of `ipfs swarm peers` is nothing.

Which steps am I missing?

Btw, the containers prints the following in regular intervals, perhaps this is related:

```
ERROR       core: no bootstrap nodes configured: go-ipfs may have difficulty connecting to the network bootstrap.go:144
```